src/LibDDRV.sol

```diff
@@ -0,0 +1,165 @@
+// SPDX-License-Identifier: MIT
+
+/*
+ * Library for generating dynamically weighted discrete probability mass function
+ * random variates.
+ *
+ * Copyright © 2023 Valorem Labs Inc.
+ *
+ * Author: 0xAlcibiades <alcibiades@valorem.xyz>
+ */
+
+pragma solidity >=0.8.8;
+
+struct Range {
+    uint256 weight;
```

Comment:
how much would we save by using a narrower int width for the weight? 

currently each Range will take up 1 slot. since they're stored in mappings, they should be packed if the struct is < 16B. So we could at least halve the # of storage slots used for the `Range`s

1 per each `Level` struct if we pack weight + roots, and 1 more if we tighten the range index ~= 2 per `Level` (not a lot, since there are very few levels).

we might consider a weight of <= 128.